Reporting feature documentation (#57)
IBM/detect-secrets: docs/audit.md

Base64HighEntropyString: True  (4.089)
...
```

## Report Generation

> Maybe you must generate a report with all of the detect-secrets findings. While similarly named, IBM's reporting feature fulfills a _different_ use case from [that of Yelp](https://github.com/Yelp/detect-secrets/blob/master/docs/audit.md#report-generation). IBM's implementation is intended be run in CI / CD pipelines for auditing purposes, or locally while testing. With it, certain checks would be executed against the results of a baseline file, and the resultant report would indicate any of those which failed.

Reviewer: ```suggestion
Want to generate a report with all the detect-secrets findings? 

While similarly named, IBM's reporting feature fulfills a _different_ use case from [that of Yelp](https://github.com/Yelp/detect-secrets/blob/master/docs/audit.md#report-generation). IBM's implementation is intended be run in CI / CD pipelines for auditing purposes, or locally while testing. With it, certain checks would be executed against the results of a baseline file, and the resultant report would indicate any of those which failed.
```
That first sentence didnt make sense to me

### Running in CI / CD

To determine whether the CI / CD stage should pass, the report will emit an exit code.

If a report is run without any `fail-on` arguments (`detect-secrets audit --report .secrets.baseline`), it will execute all the fail checks by default, yet always emit a zero exit code—even if checks fail.

In CI / CD, it is recommended to provide all `fail-on` args:

```shell
detect-secrets audit --report --fail-on-on-unaudited fail-on-live --fail-on-on-audited-real .secrets.baseline
```

> TODO: add instructions for setting up in CI / CD

Author: The docs for configuring reporting in CI / CD will be added in a later PR.
